### DISCUSSION TITLE
What are the differences between the MapLibre iOS SDK and the Mapbox iOS SDK?

### DISCUSSION THREAD

**Discussion by PeterKaminski09** [maplibre/maplibre-native#101] [Q&A]
Hi team,

I'm not too savvy on Swift Package Manager or CocoaPods, but I was wondering if someone could shed light on the difference between installing the Mapbox dependency via SPM from `https://github.com/maplibre/maplibre-gl-native-distribution` as opposed to installing v5.9 straight from the Mapbox [CocoaPods repo](https://github.com/CocoaPods/Specs/blob/master/Specs/a/5/9/Mapbox-iOS-SDK/5.9.0/Mapbox-iOS-SDK.podspec.json)

From what I can tell, the MapLibre library has the same [Podspec file](https://github.com/maplibre/maplibre-gl-native/blob/master/platform/ios/platform/ios/Mapbox-iOS-SDK.podspec) that is also pointing to the same underlying source (https://mapbox.s3.amazonaws.com/mapbox-gl-native/ios/builds/mapbox-ios-sdk-5.9.0-dynamic.zip)

A few general questions:
- Is there any difference in the underlying source code between MapLibre and Mapbox when it comes to the iOS dependency?
- Are the licenses the same?
- If there is a difference between the two, is it possible to get MapLibre as a CocoaPod instead of via SPM?

Thanks in advance for the help!

**Comment by rudolfkopriva** [comment]
Hi,

It is possible, CocoaPods support XCFramework distribution and MapLibre can be distributed with Cocoapods.
You would have to do a few modifications in the Podspec file:

Change `name` and `version`
Change source url to maplibre zip that contains the xcframework: `https://github.com/maplibre/maplibre-gl-native/releases/download/ios-v5.11.0/Mapbox-5.11.0.zip`
Change m.`vendored_frameworks` from `'dynamic/Mapbox.framework'` to `'Mapbox.xcframework'`

You can also remove MapboxMobileEvents dependency, because it is not used in MapLibre.

You can test it first with private Spec repository.

**Reply by PeterKaminski09** [reply]
Okay cool, thank you!

Do you happen to know the answer to this question?
> Is there any difference in the underlying source code between MapLibre and Mapbox when it comes to the iOS dependency?

**Reply by rudolfkopriva** [reply]
You can migrate from Mapbox 5.9.0 to MapLibre 5.11.0 without doing many changes in your app. All namespaces and class names are the same in both libs.
